File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/inventory/storage/StorageObjectsRemoveMessage.scala





















// Generated on 05/03/2013 10:34:58
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait StorageObjectsRemoveMessage extends MessageDeserializer {
  val messageId = 6035

  protected trait Message extends BaseMessage with Serializable {
    val messageId = StorageObjectsRemoveMessage.messageId

    val objectUIDList: Seq[Int]

    def serialize(buf: Buffer) {
      buf.writeUShort(objectUIDList.length)
      for (entry <- objectUIDList) {
        buf.writeInt(entry)
      }
    }

  }

  protected def create(objectUIDList: Seq[Int]): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val objectUIDList = {
      val builder = collection.immutable.Seq.newBuilder[Int]
      for (i <- (0 to buf.readUShort())) {
        builder += buf.readInt()
      }
      builder.result()
    }

    create(objectUIDList)
  }

}

object StorageObjectsRemoveMessage extends StorageObjectsRemoveMessage {
  case class MessageImpl(objectUIDList: Seq[Int]) extends Message
  type Target = MessageImpl

  override def create(objectUIDList: Seq[Int]) = MessageImpl(objectUIDList)

  def apply(objectUIDList: Seq[Int]) = create(objectUIDList)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/StorageObjectsRemoveMessage.java


// Generated on 05/08/2013 19:37:57
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class StorageObjectsRemoveMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6035;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int[] objectUIDList;
    
    public StorageObjectsRemoveMessage() { }
    
    public StorageObjectsRemoveMessage(int[] objectUIDList) {
        this.objectUIDList = objectUIDList;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeUShort(objectUIDList.length);
        for (int entry : objectUIDList) {
            buf.writeInt(entry);
        }
    }
    
    @Override
    public void deserialize(Buffer buf) {
        int limit = buf.readUShort();
        objectUIDList = new int[limit];
        for (int i = 0; i < limit; i++) {
            objectUIDList[i] = buf.readInt();
        }
    }
    
}
